sklearn/utils/estimator_checks.py

```diff
@@ -67,7 +67,14 @@
                 'RandomForestRegressor', 'Ridge', 'RidgeCV']
 
 
-def _yield_non_meta_checks(name, Estimator):
+def assert_almost_equal_dense_sparse(x, y, err_msg=''):
+    if sparse.issparse(x):
+        assert_allclose(x.data, y.data, err_msg=err_msg)
```

Comment:
should I rename it to ``data_allclose`` or change it to also check the class and sparsity pattern? That would really make the code more complex... hm I guess we could check_array and make sure it's csc or csr.